src/components/tasks/card/index.tsx

```diff
@@ -121,7 +132,51 @@ const Card: FC<Props> = ({
       </span>
       )
   }
- 
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const { requestPromise } = fetch({ url: SELF_URL });
+        const { data } = await requestPromise;
+        const userRoles = {
+          adminUser: data.roles?.admin,
+          superUser: data.roles?.super_user,
+        };
+        const { adminUser, superUser } = userRoles;
+        setIsUserAuthorized(!!adminUser || !!superUser);
+      } catch (err: any) {
+        toast(ERROR, err.message);
+      }
+    };
+    fetchData();
+
+    return (() => {
+      setIsUserAuthorized(false);
+    });
+  }, []);
+
+  function handleKeyDown(event: any): void {
```

Comment:
I believe we already know it is KeyBoardEvent right? 
Could we do more research on this
This is one ref: https://www.delftstack.com/howto/typescript/typescript-event-type/